WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -262,6 +262,64 @@ def edit
     render :edit
   end
 
+  def payment_setup
+    @competition = Competition.find(params[:id])
+
+    client = create_oauth_client
+    oauth_params = {
+      scope: 'read_write',
+      state: @competition.id,
+    }
+    @authorise_url = client.auth_code.authorize_url(oauth_params)
```

Comment:
"authorise" is as valid a spelling as "authorize", but I'd go with the Z version, since that's what the function is called.